mtg_card_identifier/CardScannerFile.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  CardScannerFile.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import UIKit
+import VisionKit
+
+struct CardScanner: UIViewControllerRepresentable {
+    
+    //    @Binding var startScanning: Bool
+    @Binding var scanedCard: [String]
+    
+    func makeCoordinator() -> Coordinator {
+        Coordinator(self)
+    }
+    
+    func makeUIViewController(context: Context) -> DataScannerViewController {
+        
+        let vc = DataScannerViewController(
+            recognizedDataTypes: [.text()],
+            qualityLevel: .balanced,
+            recognizesMultipleItems: true,
+            isHighFrameRateTrackingEnabled: true
+            //            isGuidanceEnabled: true,
+//            Don't know if we want that to work but it's basically done
+//            isHighlightingEnabled: true
+        )
+        let partialTransparentView = PartialTransparentView(cutout: CGRect(x: 31, y: 75, width: 350, height: 470))
+        partialTransparentView.translatesAutoresizingMaskIntoConstraints = true
+        partialTransparentView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        partialTransparentView.frame = vc.overlayContainerView.bounds
+        
+        vc.view.addSubview(partialTransparentView)
+        vc.regionOfInterest = CGRect(x: 31, y: 75, width: 350, height: 430)
+        vc.delegate = context.coordinator
+        
+        return vc
+    }
+    
+    func updateUIViewController(_ uiViewController: DataScannerViewController, context: Context) {
```

Comment:
This function gets called every time something in SwiftUI changes. It's meant to be the place where you update your UIViewController when SwiftUI changes or redraws. 
That means it will be called a LOT.